### ISSUE TITLE
TEMP FIXED: Prisma Client postinstall fails with Bun (Temp solution at end)

### ISSUE THREAD

**Issue by maietta** [oven-sh/bun#12266]
### What version of Bun is running?

1.1.17

### What platform is your computer?

Linux 6.9.6-200.fc40.x86_64 x86_64 Fedora 40 Workstation

### What steps can reproduce the bug?

I have setup a detailed Minimum Reproducible Example:

https://github.com/maietta/mre-sveltekit-prisma

Taken from setup.txt (included)

```The following steps were taken to set this project up:

Create project:
bun create svelte@latest
    (Choose ESLint, Prettier, Experimental Svelte 5 support)

Switch the adaptor:
bun add -d svelte-adapter-bun
    (Modify the svelte.config.js to refernce the svelte-adapter-bun)

Add Prisma CLI & Client
bun add -D prisma
bun prisma init

Run the SQL create script found in sql/ to create some sample tables.

Introspect the database to create a local Prisma schema.
bun prisma db pull

(Side note: For the purposes of this demo, we will not be baselining the database, but this would be an important step in production.)

Now we are ready for setting up our application.

Add the Prisma Client dependency and generate the local client interface, which by default is included in node_modules/
bun add @prisma/client

Generate the client interface, which will be by default, located in node_modules/, a directory that is not included in our commits.

bun prisma generate

NOTE: This is where we begin to see a problem? This might have been fixed, but we're about to find out. Running the generate command appears to work.

With all the prerequisites out of the way, let's wire up the database to the app and display something out of the database.

After dropping in the Dockerfile and attempting to build, we get hit with the following:

PrismaClientInitializationError: Prisma Client could not locate the Query Engine for runtime "debian-openssl-1.1.x".
2.462 
2.462 This happened because Prisma Client was generated for "rhel-openssl-3.0.x", but the actual deployment required "debian-openssl-1.1.x".
2.462 Add "debian-openssl-1.1.x" to `binaryTargets` in the "schema.prisma" file and run `prisma generate` after saving it:
2.462 
2.462 generator client {
2.462   provider      = "prisma-client-js"
2.462   binaryTargets = ["native", "debian-openssl-1.1.x"]
2.462 }

Let's fix that by updating the prisma.schema and re-running the generate client command.

That did not fix the problem, but now we are starting to reproduce the issues I expierenced.

See logs/ for the various log outputs after each fix attempt.

log2-after-installing-nodejs.txt details what happens when we go ahead and update our Dockerfile to include nodejs. Note that building the docker image locally no longer is successful because now we generate an error. I believe this error might be a big clue as to what is happening.

Let's now try and generate our client outside the node_modules/ and in our project instead, but I think this is a moot point, since we now have errors that seem to point to the problem.

Now that we've got our Prisma.schema with the following:

generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native", "debian-openssl-1.1.x"]
  output        = "../src/lib/server/prisma/client"
},

We need to run `bun prisma generate` once again, which will build a new client in our src/lib/server/prisma/client directory. Adjust the local reference in src/lib/prisma.ts.

Afer this, we now have a log3. "Error when evaluating SSR module /src/lib/server/prisma/client/index-browser.js: - ReferenceError: exports is not defined"

Okay, now I'm getting stumped.

Next, I try and update Dockerfile to use `npx prisma generate` instead of `bun prisma generate`. This results in same problem. This is not our issue.
```

### What is the expected behavior?

The `bun prisma generate` should just work. (The only way I was able to get an error, was to actually install nodejs in the environment. Prior to that, there were no errors produced, but the application could not build without it)

### What do you see instead?

> [builder  5/10] RUN bun install:                                                                                                                                    
0.170 bun install v1.1.17 (bb66bba1)                                                                                                                                   
0.172 Resolving dependencies                                                                                                                                           
1.224 Resolved, downloaded and extracted [2]                                                                                                                           
2.671 /app/node_modules/@prisma/client/scripts/postinstall.js:233                                                                                                      
2.671       const { js, ts } = defaultFiles[file] ?? {}
2.671                                              ^
2.671 
2.671 SyntaxError: Unexpected token '?'
2.671     at wrapSafe (internal/modules/cjs/loader.js:915:16)
2.671     at Module._compile (internal/modules/cjs/loader.js:963:27)
2.671     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
2.671     at Module.load (internal/modules/cjs/loader.js:863:32)
2.671     at Function.Module._load (internal/modules/cjs/loader.js:708:14)
2.671     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12)
2.671     at internal/main/run_main_module.js:17:47
2.671 
2.671 error: postinstall script from "@prisma/client" exited with 1
------
Dockerfile:15
--------------------
  13 |     
  14 |     COPY --link bun.lockb package.json ./
  15 | >>> RUN bun install
  16 |     
  17 |     # Generate Prisma Client
--------------------
ERROR: failed to solve: process "/bin/sh -c bun install" did not complete successfully: exit code: 1

### Additional information

The reason I filed this bug report here, is because Bun is alledgedly a drop-in replacement for NodeJS and things should "just work". However, there is an apparent difference when trying to process line 223 in Prisma Client version 5.16.1

2.671 /app/node_modules/@prisma/client/scripts/postinstall.js:233                                                                                                      
2.671       const { js, ts } = defaultFiles[file] ?? {}
2.671                                              ^

For further reference, here's 5.16.1 of the Prisma Client.

https://github.com/prisma/prisma/blob/main/packages/client/scripts/postinstall.js#L233

**Comment by maietta** [comment]
Update:

I've adjusted the Dockerfile to use node:latest as my base image, because it's a node environment Prisma is depending on.

Then I used npx prisma generate instead of Bun. Still using Bun for the runtime and dependency installs, after installing globally via npm install -g bun.

This is not a real fix, but a patch for me to get a website deployed today.

**Comment by maietta** [comment]
SOLUTION FOUND:

Apparently, Prisma Client mandates the use of Node environment. Not just any, but a more recent version, for the `prisma generate` command to work correctly.

Thanks to @Jarred-Sumner for pointing out that I was using an old version of Node, I was able to update my Dockerfile to pull down a release and unpack it and use that-- so I can specify exactly which version I want for Node.

I've asked Prisma support Bun officially, so we are not dependent on Node. It seems as though this requires only a small change on their part.

In this scenario, `bun prisma generate` works correctly.

My updated working Dockerfile for deployments now look like this:

```Dockerfile
ARG BUN_VERSION=1.1.17
FROM oven/bun:${BUN_VERSION} AS builder

# Install required dependencies
RUN apt-get update && \
    apt-get install -y \
    curl \
    xz-utils \
    && rm -rf /var/lib/apt/lists/*

# Set the Node.js version
ENV NODE_VERSION=20.15.0

# Download and install Node.js
RUN curl -fsSL https://nodejs.org/dist/v${NODE_VERSION}/node-v${NODE_VERSION}-linux-x64.tar.xz -o node-v${NODE_VERSION}-linux-x64.tar.xz && \
    tar -xf node-v${NODE_VERSION}-linux-x64.tar.xz && \
    mv node-v${NODE_VERSION}-linux-x64 /usr/local/node && \
    ln -s /usr/local/node/bin/node /usr/local/bin/node && \
    ln -s /usr/local/node/bin/npm /usr/local/bin/npm && \
    ln -s /usr/local/node/bin/npx /usr/local/bin/npx && \
    rm node-v${NODE_VERSION}-linux-x64.tar.xz

# Verify installation
RUN node --version && npm --version

WORKDIR /app

ENV NODE_ENV=production

COPY --link bun.lockb package.json ./
RUN bun install

# Generate Prisma Client
COPY --link prisma .
# RUN bunx --bun prisma generate
RUN bun prisma generate

# Copy application code
COPY --link . .

# Build application
RUN bun run build

# Remove development dependencies
RUN rm -rf node_modules && bun install --ci

FROM oven/bun:${BUN_VERSION}

# Copy Node.js binaries and curl from the builder stage
COPY --from=builder /usr/local/node /usr/local/node
COPY --from=builder /usr/bin/curl /usr/bin/curl
COPY --from=builder /usr/lib/x86_64-linux-gnu /usr/lib/x86_64-linux-gnu

# Create symbolic links to Node.js binaries
RUN ln -s /usr/local/node/bin/node /usr/local/bin/node && \
    ln -s /usr/local/node/bin/npm /usr/local/bin/npm && \
    ln -s /usr/local/node/bin/npx /usr/local/bin/npx

# Copy the built application and necessary files from the builder stage
COPY --from=builder /app /app

# Set environment variables and expose the port
ENV PORT=3000
EXPOSE 3000/tcp

# Set user and define healthcheck
USER bun
HEALTHCHECK --interval=5s --timeout=5s --start-period=5s --retries=3 CMD curl --fail http://localhost:3000/healthcheck

CMD ["bun", "build/index.js"]
```

**Comment by cholasimmons** [comment]
This is bonkers! Maybe Prisma don't want us to be using Bun.
After months of me getting problems building a docker image successfully, the Bun team released v1.1.19 (and 1.1.20) and I was finally able to `prisma generate` inside a Dockerfile.
Today I noticed prisma 5.17.0 was out, I got that update and again, 💥 I'm unable to generate 🤦🏻
